### ISSUE TITLE
Error "Task was destroyed but it is pending!" rec'd when stopped a QEMU VM

### ISSUE THREAD

**Issue by wdennis** [GNS3/gns3-server#310]
2015-09-11 15:28:12 ERROR base_events.py:912 Task was destroyed but it is pending!
task: <Task pending coro=<get() running at /usr/local/lib/python3.4/asyncio/queues.py:186> wait_for=<Future pending cb=[Task._wakeup()]> cb=[_release_waiter(<Future cancelled>)() at /usr/local/lib/python3.4/asyncio/tasks.py:333]>
2015-09-11 19:39:37 INFO qemu_vm.py:892 Stopping QEMU VM "Cumulus-VX-sw1-1" PID=1557
2015-09-11 19:39:37 INFO qemu_vm.py:876 QEMU process has stopped, return code: 0

**Comment by noplay** [comment]
Do you manage to replicate it?

**Comment by wdennis** [comment]
No, cannot seem to replicate that error on shutdown now...

**Comment by noplay** [comment]
:( I already seen that and have an hard time to figurate how to replicate . You are under linux?

**Comment by wdennis** [comment]
Yes, server (1.4b1) running on CentOS 7.1

When I got the error, the QEMU VM had been running for a long time (a few
days); my tests to see if I get the error again, I have only run the VM for
a minute or two...

**Comment by wdennis** [comment]
Played around with more of the QEMU VM stuff in GNS3 tonight; I saw that a connection between the two QEMU VMs was failing (will further test and open issue if warranted) but when I closed the project, I got a warning that a UDP port was still in use. So I issued a Ctrl-C to shut down the gns3server (for now I start it in the foreground in a tmux session) and got the "Task was destroyed" error again as it shut down:

'''
2015-09-15 19:43:37 INFO qemu_vm.py:892 Stopping QEMU VM "Cumulus-VX-port1402" PID=2911
2015-09-15 19:43:37 INFO qemu_vm.py:892 Stopping QEMU VM "Cumulus-VX-port1401" PID=2912
2015-09-15 19:43:37 INFO qemu_vm.py:876 QEMU process has stopped, return code: 0
2015-09-15 19:43:37 INFO qemu_vm.py:876 QEMU process has stopped, return code: 0
2015-09-15 20:05:05 INFO qemu_vm.py:533 QEMU VM "Cumulus-VX-port1402" [be532e38-481a-4aaa-acb9-18ef822f3b89] has disabled ACPI shutdown
2015-09-15 20:05:05 INFO qemu_vm.py:533 QEMU VM "Cumulus-VX-port1401" [bc508c20-a60e-42ce-9ade-cd96c9da67be] has disabled ACPI shutdown
2015-09-15 20:05:05 INFO project.py:380 Project 535c0f6c-bd3d-4b2a-ad29-c65d0cfefc9f with path '/home/gns3user/GNS3/projects/535c0f6c-bd3d-4b2a-ad29-c65d0cfefc9f' closed
2015-09-15 20:05:05 WARNING project.py:385 Project 535c0f6c-bd3d-4b2a-ad29-c65d0cfefc9f has UDP ports still in use: {10002}
^C2015-09-15 20:05:41 WARNING server.py:108 Server has got signal SIGINT, exiting...
2015-09-15 20:05:41 ERROR base_events.py:912 Task was destroyed but it is pending!
task: <Task pending coro=<get() running at /usr/local/lib/python3.4/asyncio/queues.py:186> wait_for=<Future pending cb=[Task._wakeup()]> cb=[_release_waiter(<Future cancelled>)() at /usr/local/lib/python3.4/asyncio/tasks.py:333]>
2015-09-15 20:05:41 ERROR base_events.py:912 Task was destroyed but it is pending!
task: <Task pending coro=<get() done, defined at /usr/local/lib/python3.4/asyncio/queues.py:160> wait_for=<Future pending cb=[Task._wakeup()]> cb=[_release_waiter(<Future cancelled>)() at /usr/local/lib/python3.4/asyncio/tasks.py:333]>
'''

**Comment by wdennis** [comment]
Looks like I get it every time I shut down the gns3server process after playing with QEMU...

This on a band-net topology:

```
2015-09-15 20:16:48 INFO qemu_vm.py:892 Stopping QEMU VM "Cumulus-VX-port1402-1" PID=3451
2015-09-15 20:16:48 INFO qemu_vm.py:892 Stopping QEMU VM "Cumulus-VX-port1401-1" PID=3458
2015-09-15 20:16:48 INFO qemu_vm.py:876 QEMU process has stopped, return code: 0
2015-09-15 20:16:48 INFO qemu_vm.py:876 QEMU process has stopped, return code: 0
2015-09-15 20:16:57 INFO qemu_vm.py:533 QEMU VM "Cumulus-VX-port1401-1" [2bb07a7b-4d6d-4379-97ce-54c138d5f01e] has disabled ACPI shutdown
2015-09-15 20:16:57 INFO qemu_vm.py:533 QEMU VM "Cumulus-VX-port1402-1" [f35cd3ce-fcb9-4fe1-84ec-631edf37b48a] has disabled ACPI shutdown
2015-09-15 20:16:57 INFO project.py:380 Project e1caa8ec-06a2-44e3-aca0-7af8bb02272b with path '/home/gns3user/GNS3/projects/e1caa8ec-06a2-44e3-aca0-7af8bb02272b' closed
^C2015-09-15 20:17:04 WARNING server.py:108 Server has got signal SIGINT, exiting...
2015-09-15 20:17:04 ERROR base_events.py:912 Task was destroyed but it is pending!
task: <Task pending coro=<get() done, defined at /usr/local/lib/python3.4/asyncio/queues.py:160> wait_for=<Future pending cb=[Task._wakeup()]> cb=[_release_waiter(<Future cancelled>)() at /usr/local/lib/python3.4/asyncio/tasks.py:333]>
```

**Comment by noplay** [comment]
Thanks for the details

**Comment by noplay** [comment]
I think I have fixed the errors in 1.3.12

**Comment by noplay** [comment]
By the way I discover process priority in qemu was broken
